File: backup/capacitor/ios/App/App/utils/ConverterUtils.swift
//
//  ConverterUtils.swift
//  App
//
//  Created by zetor on 2018. 07. 03..
//

import Foundation
import UIKit

public class ConvertUtils {
    public static func convert(bound:[String:Double]!, zoomScale : CGFloat, deviation : CGFloat = 0.0) -> CGRect? {
        let x = CGFloat(bound["left"]!) * CGFloat(zoomScale);
        let y = CGFloat(bound["top"]!) * CGFloat(zoomScale) + deviation;
        let width = CGFloat(bound["right"]!) * CGFloat(zoomScale) - x;
        let height = CGFloat(bound["bottom"]!) * CGFloat(zoomScale) - y;
        return CGRect(x:x, y:y, width:width, height: height);
    }
}

File: backup/capacitor/android/app/src/main/java/com/raxim/capacitor/utils/ConverterUtils.java
package com.raxim.capacitor.utils;

import android.graphics.Rect;
import android.graphics.RectF;
import android.support.design.widget.CoordinatorLayout;
import android.view.View;

import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.MarkerOptions;

import org.json.JSONException;
import org.json.JSONObject;

public class ConverterUtils {
    public static RectF convert(JSONObject jsonObject, float zoomScale) throws JSONException {
        float left = Float.parseFloat(jsonObject.getString("left")) * zoomScale;
        float right = Float.parseFloat(jsonObject.getString("right")) * zoomScale;
        float top = Float.parseFloat(jsonObject.getString("top")) * zoomScale;
        float bottom = Float.parseFloat(jsonObject.getString("bottom")) * zoomScale;
        return new RectF(left, top, right, bottom);
    }

    public static MarkerOptions convert(JSONObject jsonObject) throws JSONException {
        Double lat = jsonObject.getDouble("lat");
        Double lng = jsonObject.getDouble("lng");
        MarkerOptions markerOptions = new MarkerOptions();
        markerOptions.position(new LatLng(lat, lng));
        return markerOptions;
    }

    public static RectF locateView(View view) {
        RectF loc = new RectF();
        CoordinatorLayout.LayoutParams layoutParams = (CoordinatorLayout.LayoutParams) view.getLayoutParams();

        loc.left = layoutParams.leftMargin;
        loc.top = layoutParams.topMargin;
        loc.right = layoutParams.leftMargin + layoutParams.width;
        loc.bottom = layoutParams.topMargin + layoutParams.height;
        return loc;
    }
}
